libbpf-rs/src/btf/types.rs

```diff
@@ -863,26 +844,333 @@ gen_collection_concrete_type! {
+            BtfKind::Void => "",
+            BtfKind::Ptr => "",
+            BtfKind::Array => "",
+            BtfKind::Union => "",
+            BtfKind::Enum => "",
+            BtfKind::Fwd => "",
+            BtfKind::Typedef => "",
+            BtfKind::Volatile => "",
+            BtfKind::Const => "",
+            BtfKind::Restrict => "",
+            BtfKind::Func => "",
+            BtfKind::FuncProto => "",
+            BtfKind::Var => "",
+            BtfKind::DataSec => "",
+            BtfKind::Float => "",
+            BtfKind::DeclTag => "",
+            BtfKind::TypeTag => "",
+            BtfKind::Enum64 => "",
+        });
+    }
+
+    #[test]
+    fn partial_match() {
+        dummy_type!(ty);
+        btf_type_match!(match ty {
+            BtfKind::Int => "int",
+            _ => "default",
+        });
+    }
+
+    #[test]
+    fn or_pattern_match() {
+        dummy_type!(ty);
+        // we ask rustfmt to not format this block so that we can keep the trailing `,` in the int
+        // branch.
+        #[rustfmt::skip]
+        btf_type_match!(match ty {
+            BtfKind::Int => "int",
+            BtfKind::Struct | BtfKind::Union => "composite",
+            BtfKind::Typedef | BtfKind::Volatile => {
+                "qualifier"
+            }
+            BtfKind::Const | BtfKind::Restrict => {
+                "const or restrict"
+            },
+            _ => "default",
+        });
+    }
+
+    #[test]
+    fn match_arm_with_brackets() {
+        dummy_type!(ty);
+        // we ask rustfmt to not format this block so that we can keep the trailing `,` in the int
+        // branch.
+        #[rustfmt::skip]
+        btf_type_match!(match ty {
+            BtfKind::Void => {
+                "void"
+            }
+            BtfKind::Int => {
+                "int"
+            },
+            BtfKind::Struct => "struct",
+            _ => "default",
+        });
+    }
+
+    #[test]
+    fn match_on_composite() {
+        dummy_type!(ty);
+        btf_type_match!(match ty {
+            BtfKind::Composite(c) => c.is_struct,
+            _ => false,
+        });
+        btf_type_match!(match ty {
+            BtfKind::Composite(c) => {
+                c.is_struct
+            }
+            _ => false,
+        });
+        // we ask rustfmt to not format this block so that we can keep the trailing `,` in the int
+        // branch.
+        #[rustfmt::skip]
+        btf_type_match!(match ty {
+            BtfKind::Composite(c) => {
+                c.is_struct
+            },
+            _ => false,
+        });
+    }
+
+    #[test]
+    fn match_arm_with_multiple_statements() {
+        dummy_type!(ty);
+
+        match ty.kind() {
+            crate::btf::BtfKind::Int => {
+                let i = crate::btf::types::Int::try_from(ty).unwrap();
+                std::hint::black_box(i);
+                "int"
+            }
+            _ => {
+                std::hint::black_box(());
```

Comment:
What purpose does this `black_box` call have here? It's not clear to me why we need it.

Reply:
I didn't want to use a println but wanted something that looked like a statement. I guess I could have used a simple assignment
```rs
let x = i;
x
```
instead of being all fancy with the black_box.

TL;DR: It serves no special purpose besides being a statement